### DISCUSSION TITLE
Parallel reading/writing of single parquet files?

### DISCUSSION THREAD

**Discussion by bradlarsen** [apache/arrow-rs#6293] [Q&A] [answered]
Hi there,

The C++ implementation of Arrow seems to support multithreaded reading of Parquet files (e.g., [here](https://arrow.apache.org/docs/cpp/api/formats.html#_CPPv4NK7parquet21ArrowReaderProperties11use_threadsEv)).

It doesn't look like the native Rust implementation has any multithreaded capability like this. Are there any plans or ongoing efforts to add that?

Thanks!

**Comment by tustvold** [comment, marked as answer]
See https://docs.rs/parquet/latest/parquet/arrow/arrow_writer/struct.ArrowColumnWriter.html

And 

https://docs.rs/parquet/latest/parquet/arrow/async_reader/type.ParquetRecordBatchStreamBuilder.html#method.new_with_metadata
